kork-retrofit/src/test/java/com/netflix/spinnaker/kork/retrofit/exceptions/RetrofitExceptionTest.java

```diff
@@ -97,4 +110,336 @@ public void testServerException() {
+
+    // add the ByteArrayConverterFactoryTestUtil to convert response body to bytes.
+    Retrofit customRetrofit =
+        new Retrofit.Builder()
+            .baseUrl("http://localhost:8987/")
+            .addConverterFactory(ByteArrayConverterFactoryTestUtil.create())
+            .build();
+
+    // get the retrofit2 http exception from the actual response
+    RetrofitException retrofitException =
+        RetrofitException.httpError(actualResponse, customRetrofit);
+    assertNull(retrofitException.getCause());
+    assertEquals(retrofitException.getMessage(), "401 Response.error()");
+
+    // hashmap for json body
+    // Without the ByteArrayConverterFactoryTestUtil, and with only GsonConverterFactory,
+    // the getBodyAs() will give a com.fasterxml.jackson.core.JsonParseException
+    byte[] actualResponseBodyContent = retrofitException.getBodyAs(byte[].class);
+    assertNotNull(actualResponseBodyContent);
+    assertEquals(actualResponseBodyContent.length, byteArray.length);
+
+    for (int i = 0; i < byteArray.length; i++) {
+      assertEquals(byteArray[i], actualResponseBodyContent[i]);
+    }
+  }
+
+  private void testApiServiceForJsonResponse(Map expectedContentBodyMap, String jsonContentString)
+      throws IOException {
+    // default converter is gson when factory list is null
+    this.testApiServiceForJsonResponse(expectedContentBodyMap, jsonContentString, null);
+  }
+
+  private void testApiServiceForJsonResponse(
+      Map expectedContentBodyMap,
+      String jsonContentString,
+      List<Converter.Factory> converterFactoryList)
+      throws IOException {
+    // Here we have randomly picked 401 - UNAUTHORIZED error.
+    // However, our main intent is to check the RetrofitException behaviour for different response
+    // bodies.
+
+    // validate contents cannot be null, otherwise it gives NullPointerException
+    assertNotNull(jsonContentString);
+
+    String invalidContentJsonBodyString = (String) jsonContentString;
+    // validate contents cannot be null, otherwise it gives NullPointerException
+    ResponseBody responseBody =
+        ResponseBody.create(
+            MediaType.parse("application/json" + "; charset=utf-8"), invalidContentJsonBodyString);
+
+    // validate responseBody cannot be null, otherwise it gives NullPointerException
+    assertNotNull(responseBody);
+
+    Response<?> actualResponse = Response.error(HttpStatus.UNAUTHORIZED.value(), responseBody);
+    assertNotNull(actualResponse);
+
+    // assert response body is null
+    String actualBody = (String) actualResponse.body();
+    assertNull(actualBody);
+
+    // Assert the error body is there
+    ResponseBody errorBody = actualResponse.errorBody();
+    assertNotNull(errorBody);
+
+    // create a builder
+    Retrofit.Builder customRetrofitBuilder =
+        new Retrofit.Builder().baseUrl("http://localhost:8987/");
+
+    // Use default as GsonConverterFactory if not specified,  for json list with lenient option to
```

Comment:
```suggestion
    // Use GsonConverterFactory if not specified,  for json list with lenient option to
```
Please also explain the choice of GsonConverterFactory (it may be arbitrary).  Is it the only one with a lenient option?